src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.CreateToken.cs

```diff
@@ -693,26 +697,34 @@ int sizeOfEncodedHeaderAndPayloadAsciiBytes
             bool setDefaultTimesOnTokenCreation,
             int tokenLifetimeInMinutes)
         {
-            bool audienceChecked = false;
+            bool descriptorClaimsAudienceChecked = false;
```

Comment:
Changing these names adds questionable value, but increases the PR review time and increases the probability of a conflict.
Please reset to original.

Reply:
I believe they add clarity that is otherwise difficult to obtain. 

Audience (and the other claims defined here) can be defined in three different places in a SecurityTokenDescriptor object and the code dealing with this intersection of claims is spread across two methods and over 200 lines of code. However, the variables renamed only refer to claims defined in one of the three areas (SecurityTokenDescriptor.Claims), using them elsewhere breaks the logic. Additionally, the renamed variables are defined next to similar variables like audienceSet which are used for claims defined in any of the three locations making it unintuitive that a variable named audienceChecked should only be used in reference to audiences defined in one area when the code checks for claims in all three locations. That's why the change is adding the specific location those variables are intended to interact with.

For impact, this particular section of code was difficult to understand as it hinges on a few lines out of over 200, and I had to re-learn it a couple of times while working on this PR. Additionally, the confusion around the intent of these "Check" variables specifically caused me to accidentally create a bug that ended up costing hours of time. All told I could have easily saved more than half a day of work if the code was easier to comprehend. Changing the name of the variables helps address this while not changing any existing functionality or logic and imho is worth a hypothetical merge conflict or two which take maybe 5 mins to address.

Reply:
Please revert the name back to the original, and let's try to keep the PR scoped to addressing the original issue. If you want to propose a name change, that can come as a follow-up PR. Thank you.

Reply:
changed them back